Corrupted image crashes the app

Hi!

Reading image with flutter_image_compress plugin crashes the app if the file is corrupted. It would be better if plugin could handle file read failures gracefully without crashing unexpectedly.

````
java.lang.Error: java.io.IOException: read failed: EIO (I/O error)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1173)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
        at java.lang.Thread.run(Thread.java:919)
     Caused by: java.io.IOException: read failed: EIO (I/O error)
        at libcore.io.IoBridge.read(IoBridge.java:523)
        at java.io.FileInputStream.read(FileInputStream.java:313)
        at kotlin.io.FilesKt__FileReadWriteKt.readBytes(FileReadWrite.kt:70)
        at com.example.flutterimagecompress.core.CompressFileHandler$handleGetFile$1.run(CompressFileHandler.kt:85)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641) 
        at java.lang.Thread.run(Thread.java:919) 
     Caused by: android.system.ErrnoException: read failed: EIO (I/O error)
        at libcore.io.Linux.readBytes(Native Method)
        at libcore.io.Linux.read(Linux.java:190)
        at libcore.io.ForwardingOs.read(ForwardingOs.java:177)
        at libcore.io.BlockGuardOs.read(BlockGuardOs.java:303)
        at libcore.io.ForwardingOs.read(ForwardingOs.java:177)
        at libcore.io.IoBridge.read(IoBridge.java:513)
        at java.io.FileInputStream.read(FileInputStream.java:313) 
        at kotlin.io.FilesKt__FileReadWriteKt.readBytes(FileReadWrite.kt:70) 
        at com.example.flutterimagecompress.core.CompressFileHandler$handleGetFile$1.run(CompressFileHandler.kt:85) 
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167) 
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641) 
        at java.lang.Thread.run(Thread.java:919)
`

--- a/packages/flutter_image_compress_common/android/src/main/kotlin/com/fluttercandies/flutter_image_compress/core/CompressFileHandler.kt
+++ b/packages/flutter_image_compress_common/android/src/main/kotlin/com/fluttercandies/flutter_image_compress/core/CompressFileHandler.kt
@@ -32,20 +32,20 @@ class CompressFileHandler(private val call: MethodCall, result: MethodChannel.Re
                 reply(null)
                 return@execute
             }
-            val exifRotate = if (autoCorrectionAngle) {
-                val bytes = File(filePath).readBytes()
-                Exif.getRotationDegrees(bytes)
-            } else {
-                0
-            }
-            if (exifRotate == 270 || exifRotate == 90) {
-                val tmp = minWidth
-                minWidth = minHeight
-                minHeight = tmp
-            }
-            val targetRotate = rotate + exifRotate
             val outputStream = ByteArrayOutputStream()
             try {
+                val exifRotate = if (autoCorrectionAngle) {
+                    val bytes = File(filePath).readBytes()
+                    Exif.getRotationDegrees(bytes)
+                } else {
+                    0
+                }
+                if (exifRotate == 270 || exifRotate == 90) {
+                    val tmp = minWidth
+                    minWidth = minHeight
+                    minHeight = tmp
+                }
+                val targetRotate = rotate + exifRotate
                 formatHandler.handleFile(
                     context,
                     filePath,
@@ -79,11 +79,6 @@ class CompressFileHandler(private val call: MethodCall, result: MethodChannel.Re
             val targetPath = args[4] as String
             val rotate = args[5] as Int
             val autoCorrectionAngle = args[6] as Boolean
-            val exifRotate = if (autoCorrectionAngle) {
-                Exif.getRotationDegrees(File(file))
-            } else {
-                0
-            }
             val format = args[7] as Int
             val keepExif = args[8] as Boolean
             val inSampleSize = args[9] as Int
@@ -94,14 +89,19 @@ class CompressFileHandler(private val call: MethodCall, result: MethodChannel.Re
                 reply(null)
                 return@execute
             }
-            if (exifRotate == 270 || exifRotate == 90) {
-                val tmp = minWidth
-                minWidth = minHeight
-                minHeight = tmp
-            }
-            val targetRotate = rotate + exifRotate
             var outputStream: OutputStream? = null
             try {
+                val exifRotate = if (autoCorrectionAngle) {
+                    Exif.getRotationDegrees(File(file))
+                } else {
+                    0
+                }
+                if (exifRotate == 270 || exifRotate == 90) {
+                    val tmp = minWidth
+                    minWidth = minHeight
+                    minHeight = tmp
+                }
+                val targetRotate = rotate + exifRotate
                 outputStream = File(targetPath).outputStream()
                 formatHandler.handleFile(
                     context,

